nixos/modules/misc/ids.nix

```diff
@@ -352,6 +352,7 @@ in
       moonraker = 320;
       distcc = 321;
       webdav = 322;
+      pipewire = 323;
```

Comment:
Do we really need a static id?

Reply:
How to have dynamic IDs with socket units? I have tried it but couldn't get it to work. But totally open for suggestions!

Reply:
Does not `DynamicUser` work?

Reply:
No. At least I haven't succeeded doing so. Here's what I've tried:

```diff
diff --git a/nixos/modules/misc/ids.nix b/nixos/modules/misc/ids.nix
index edd75883df9..3aee49b8c57 100644
--- a/nixos/modules/misc/ids.nix
+++ b/nixos/modules/misc/ids.nix
@@ -351,7 +351,7 @@ in
       hqplayer = 319;
       moonraker = 320;
       distcc = 321;
-      pipewire = 322;
+      #pipewire = 322; # DynamicUser = true
 
       # When adding a uid, make sure it doesn't match an existing gid. And don't use uids above 399!
 
diff --git a/nixos/modules/services/desktops/pipewire/pipewire-media-session.nix b/nixos/modules/services/desktops/pipewire/pipewire-media-session.nix
index f94630552b0..b427ef99d3c 100644
--- a/nixos/modules/services/desktops/pipewire/pipewire-media-session.nix
+++ b/nixos/modules/services/desktops/pipewire/pipewire-media-session.nix
@@ -102,6 +102,8 @@ in {
     systemd.services.pipewire-media-session.wantedBy = [ "pipewire.service" ];
     systemd.user.services.pipewire-media-session.wantedBy = [ "pipewire.service" ];
 
+    systemd.services.pipewire-media-session.serviceConfig.DynamicUser = true;
+
     environment.etc."pipewire/media-session.d/media-session.conf" = {
       source = json.generate "media-session.conf" configs.media-session;
     };
diff --git a/nixos/modules/services/desktops/pipewire/pipewire.nix b/nixos/modules/services/desktops/pipewire/pipewire.nix
index 64c99a3fc9a..63552635c29 100644
--- a/nixos/modules/services/desktops/pipewire/pipewire.nix
+++ b/nixos/modules/services/desktops/pipewire/pipewire.nix
@@ -178,6 +178,13 @@ in {
     systemd.user.sockets.pipewire.wantedBy = lib.mkIf cfg.socketActivation [ "sockets.target" ];
     systemd.user.sockets.pipewire-pulse.wantedBy = lib.mkIf (cfg.socketActivation && cfg.pulse.enable) ["sockets.target"];
 
+    systemd.sockets.pipewire.socketConfig.DynamicUser = true;
+    systemd.services.pipewire.serviceConfig.DynamicUser = true;
+    systemd.services.pipewire.serviceConfig.SupplementaryGroups = [
+      "audio"
+      "video"
+    ] ++ lib.optional config.security.rtkit.enable "rtkit";
+
     services.udev.packages = [ cfg.package ];
 
     # If any paths are updated here they must also be updated in the package test.
@@ -222,16 +229,6 @@ in {
       lib.optional cfg.jack.enable "${cfg.package.jack}/lib";
 
     users = lib.mkIf cfg.systemWide {
-      users.pipewire = {
-        uid = config.ids.uids.pipewire;
-        group = "pipewire";
-        extraGroups = [
-          "audio"
-          "video"
-        ] ++ lib.optional config.security.rtkit.enable "rtkit";
-        description = "Pipewire system service user";
-        isSystemUser = true;
-      };
       groups.pipewire.gid = config.ids.gids.pipewire;
     };
```

This builds, but starting sockets fails with:
```
pipewire.socket: Failed to resolve user pipewire: No such process
pipewire.socket: Control process exited, code=exited, status=217/USER
pipewire.socket: Failed with result 'exit-code'.
Failed to listen on Multimedia System.
```

Trying to set `systemd.sockets.pipewire.serviceConfig.DynamicUser` fails to evaluate with:
```
The option `systemd.sockets.pipewire.serviceConfig' does not exist. Definition values:
- In `/path/to/nixpkgs/nixos/modules/services/desktops/pipewire/pipewire.nix':
    {
      DynamicUser = true;
    }
```